src/main/java/examples/CassandraClientExamples.java

```diff
@@ -0,0 +1,177 @@
+            if (resultSet.isFullyFetched()) {
+              System.out.println("The result is fully fetched, we don't need to call this method for one more time!");
+            } else {
+              System.out.println("The result still does not fully fetched");
+            }
+          } else {
+            System.out.println("Unable to fetch more results");
+            fetchMoreResults.cause().printStackTrace();
+          }
+        });
+
+      } else {
+        System.out.println("Unable to execute the query");
+        execute.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void streaming(CassandraClient cassandraClient) {
+    cassandraClient.queryStream("SELECT * FROM my_keyspace.my_table where my_key = 'my_value'", queryStream -> {
+      if (queryStream.succeeded()) {
+        CassandraRowStream stream = queryStream.result();
+        stream.handler(row -> {
+          // handler will be called with row
+        });
+
+        // pausing consumption
+        stream.pause();
+
+        //resuming consumption
+        stream.resume();
+
+        stream.endHandler(end -> {
+          System.out.println("Steam of rows is ended");
+        });
+
+        stream.exceptionHandler(exception -> {
+          System.out.println("Some exception occurred");
+          exception.printStackTrace();
+        });
+
+      } else {
+        System.out.println("Unable to execute the query");
+        queryStream.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void fetchAll(CassandraClient cassandraClient) {
+    cassandraClient.executeWithFullFetch("SELECT * FROM my_keyspace.my_table where my_key = 'my_value'", executeWithFullFetch -> {
+      if (executeWithFullFetch.succeeded()) {
+        List<Row> rows = executeWithFullFetch.result();
+        for (Row row : rows) {
+          // handle each row here
+        }
+      } else {
+        System.out.println("Unable to execute the query");
+        executeWithFullFetch.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void prepareQuery(CassandraClient cassandraClient) {
+    cassandraClient.prepare("SELECT * FROM my_keyspace.my_table where my_key = ? ", preparedStatementResult -> {
+      if (preparedStatementResult.succeeded()) {
+        System.out.println("The query has successfully been prepared");
+        PreparedStatement preparedStatement = preparedStatementResult.result();
+        // now you can use this PreparedStatement object for the next queries
+      } else {
+        System.out.println("Unable to prepare the query");
+        preparedStatementResult.cause().printStackTrace();
+      }
+    });
+  }
+
+  public void usingPreparedStatementFuture(CassandraClient cassandraClient, PreparedStatement preparedStatement) {
+    // You can execute you prepared statement using any way to execute queries.
+
+    // Low level fetch API
+    cassandraClient.execute(preparedStatement.bind("my_value"), done -> {
+      // handle results here
```

Comment:
Can you add code to show that the async result is a `ResultSet`?

Reply:
Yeah, done

Reply:
👍